File: 2sum/TwoSum.py
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash={}
        index=0
        for val in nums:
            x=target-val
            if x in hash:
                return[index,hash[x]]
            else:
                hash[val]=index
                index+=1

File: TwoSum/TwoSum.java

import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
     Map<Integer,Integer> map = new HashMap<>();
    for (int i =0; i<nums.length;i++){
        int val = target -nums[i];
        if (map.containsKey(val)){
            int[] result = new int[2];
            result[0]=i;
            result[1]=map.get(val);
            return result;
        }else{
            map.put(nums[i],i);
        }
    }
     return new int[0];
    }
}